nima_core/embeddings.py

```
@@ -107,47 +107,42 @@ def embed_batch(texts: list[str]) -> list[list[float]]:
+            results_list.append([0.0] * EMBEDDING_DIM)
+            continue
+
+        payload = json.dumps(
+            {"model": EMBEDDING_MODEL, "prompt": txt}
+        ).encode("utf-8")
+        req = urllib.request.Request(
+            _EMBED_URL,
+            data=payload,
+            headers={"Content-Type": "application/json"},
+            method="POST",
         )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)